newIDE/app/src/AssetStore/index.js

```diff
@@ -80,6 +90,37 @@ export const AssetStore = ({ project }: Props) => {
   ] = React.useState<?PrivateAssetPackListingData>(null);
   const { onPurchaseSuccessful } = React.useContext(AuthenticatedUserContext);
 
+  // The saved scroll position must not be reset by a scroll event until it
+  // has been applied.
+  const hasAppliedSavedScrollPosition = React.useRef<boolean>(false);
+  // The saved scroll position must not be applied after users have scrolled.
+  const hasScrolled = React.useRef<boolean>(false);
+  const needScrollUpdate = React.useCallback(() => {
+    hasAppliedSavedScrollPosition.current = false;
+    hasScrolled.current = false;
+  }, []);
+
+  React.useLayoutEffect(
+    () => {
+      const scrollPosition = navigationState.getCurrentPage().scrollPosition;
+      if (scrollPosition != null && scrollView && !hasScrolled.current) {
+        scrollView.scrollToPosition(scrollPosition);
+      }
+      hasAppliedSavedScrollPosition.current = true;
+    },
+    [isOnHomePage, navigationState, openedAssetShortHeader, scrollView]
+  );
```

Comment:
I did a new version that uses call-backs to get the current scroll position.
The scroll position is applied back in a `useLayoutEffect` to have the right ref to the scrolled views.

At the 1st render, it has no effect on the scroll. It only scroll when the view is rendered again.
It works for the home page and the asset list because the view is always refreshed for some reason, but the asset detail doesn't refresh so the scroll is not applied.